File: php/features.php
<?php
declare(strict_types=1);

// Civitai SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class CivitaiFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new CivitaiBaseFeature();
            case "test":
                return new CivitaiTestFeature();
            default:
                return new CivitaiBaseFeature();
        }
    }
}

File: py/features.py
# Civitai SDK feature factory

from feature.base_feature import CivitaiBaseFeature
from feature.test_feature import CivitaiTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CivitaiBaseFeature(),
        "test": lambda: CivitaiTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
